### PR TITLE
Add godox linter and resolve trivial TODO comments

### PR DESCRIPTION
**Pull request by stana-miric**
This PR includes:
- Adding godox linter for detecting unresolved TODOs and FIXMEs
- Resolving trivial TODOs
- Marking TODOs which are going to be resolved in newly created tasks with nolint 
Tasks created for unsresolved TODOs are: EVM-519, EVM-521, EVM-522, EVM-523, EVM-524, EVM-525, EVM-527, EVM-528, EVM-529, EVM-540, EVM-541, EVM-542

# Changes include

- [ ] Bugfix (non-breaking change that solves an issue)
- [ ] Hotfix (change that solves an urgent issue, and requires immediate attention)
- [ ] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (change that is not backwards-compatible and/or changes current functionality)

# Checklist

- [x] I have assigned this PR to myself
- [x] I have added at least 1 reviewer
- [ ] I have added the relevant labels
- [ ] I have updated the official documentation
- [ ] I have added sufficient documentation in code

## Testing

- [x] I have tested this code with the official test suite
- [ ] I have tested this code manually

# Additional comments

Note that ValidatorSetAddr and StateReceiverAddr are removed form PolyBFTConfig, since we have those addresses in the contracts package. As I discuss it with Stefan, it should not affect backward compatibility since they are not included in the blockchain

### CODE REVIEW

File: network/server.go [hunk 1]
```diff
@@ -617,15 +618,19 @@ func (s *Server) NewProtoConnection(protocol string, peerID peer.ID) (*rawGrpc.C
 		return nil, err
 	}
 
-	return p.Client(stream), nil
+	if clentConn, err := p.Client(stream); err != nil {
+		return nil, err
+	} else {
+		return clentConn, nil
+	}
```
Comment (Stefan-Ethernal) [inline network/server.go]:
```suggestion
	return p.Client(stream)
```

File: network/grpc/grpc.go [hunk 2]
```diff
@@ -124,18 +124,15 @@ func (g *GrpcStream) Close() error {
...
-	conn, err := grpc.Dial("", grpc.WithTransportCredentials(insecure.NewCredentials()), opts)
-
-	if err != nil {
-		// TODO: this should not fail at all
-		panic(err) //nolint:gocritic
+	if conn, err := grpc.Dial("", grpc.WithTransportCredentials(insecure.NewCredentials()), opts); err != nil {
+		return nil, err
+	} else {
+		return conn, nil
 	}
```
Comment (Stefan-Ethernal) [inline network/grpc/grpc.go]:
```suggestion
	return grpc.Dial("", grpc.WithTransportCredentials(insecure.NewCredentials()), opts)
```

File: consensus/polybft/fsm_test.go [hunk 3]
```diff
@@ -62,7 +62,7 @@ func TestFSM_verifyCommitEpochTx(t *testing.T) {
 	t.Parallel()
 
 	fsm := &fsm{
-		config:           &PolyBFTConfig{ValidatorSetAddr: contracts.ValidatorSetContract},
+		config:           &PolyBFTConfig{},
```
Comment (Stefan-Ethernal) [inline consensus/polybft/fsm_test.go]: Also worth checking do we need the `PolyBFTConfig` reference set in other fsm unit tests.
```suggestion
```

File: e2e-polybft/e2e/helpers_test.go [hunk 4]
```diff
@@ -98,3 +99,24 @@ func getRootchainValidators(relayer txrelayer.TxRelayer, checkpointManagerAddr e
 
 	return validators, nil
 }
+
```
Comment (Stefan-Ethernal) [inline e2e-polybft/e2e/helpers_test.go:110]: Worth considering moving everything which are not e2e tests themselves from `bridge_test` here in helpers_test.go (namely: `sendExitTransaction`, `getExitProof` and `checkStateSyncResultLogs`)

**Review by Stefan-Ethernal — APPROVED** [review]
Generally LGTM, nice job!

File: network/server.go (hunk 1 above)
Comment (stana-miric) [reply network/server.go]: fix 17429bb

File: network/grpc/grpc.go (hunk 2 above)
Comment (stana-miric) [reply network/grpc/grpc.go]: fix 17429bb

File: consensus/polybft/fsm_test.go (hunk 3 above)
Comment (stana-miric) [reply consensus/polybft/fsm_test.go]: fix 17429bb

File: e2e-polybft/e2e/helpers_test.go (hunk 4 above)
Comment (stana-miric) [reply e2e-polybft/e2e/helpers_test.go:110]: fix 17429bb

**Review by vcastellm — APPROVED** [review]

**Review by goran-ethernal — APPROVED** [review]

**Review by Nemanja0x — APPROVED** [review]

**Pull request merged**